### ISSUE TITLE
Question: More Robust SpeechToTextProvider example?

### ISSUE THREAD

**Issue by cswkim** [csdcorp/speech_to_text#169]
I see the snippet [here](https://pub.dev/documentation/speech_to_text/latest/speech_to_text_provider/SpeechToTextProvider-class.html) but I'm still a bit confused as to how I may be able to implement a fairly simple scenario:

Let's say I have two separate widgets, one is an IconButton and the other is just for simplicities sake a `Container` with a `Text` widget. I'd like the button widget file to trigger listening on/off, plus depending on the `isListening` state change the Icon that is displayed. The other widget file just listens for a recognition event and just renders the output in the `Text('${wordsGoHere}')` widget.

Should both these widgets be `StatefulWidget` with local state that gets set during SpeechToTextProvider subscription events? Do you call `available = await speechProvider.initialize();` in both files after instantiating `SpeechToTextProvider`? Should you call `.stop()` in a `dispose`?

I'm not sure how to begin setting this all up. Also, what does the following do in the current example:
```
speechProvider.addListener(() {
  var words = speechProvider.lastWords;
});
```

**Comment by ab36245** [comment]
There are two ways to use SpeechToText, each of which have robust (fully working) examples.

First, using the older callback style. There is a complete and functioning example directly available on pub.dev. Look at the examples for either the default version ([3.2](https://pub.dev/packages/speech_to_text/example)) or null-safe version ([4.1.0-nullsafety](https://pub.dev/packages/speech_to_text/versions/4.1.0-nullsafety/example)).

Second, using the newer SpeechProvider interface. This is less well documented (in fact it's on my list to ask about this interface vs the older one re best practices) but there is an example available in speech_to_text/example/lib/provider_example.dart in the [github repo](https://github.com/csdcorp/speech_to_text).

Combined with following the Using section of the Readme on pub.dev either of these examples get you a complete working app on either Android or iOS. Reading through either example should answer almost all of your questions.

Personally, I use the older callback style of interface as I find this gives me all the control I want, it doesn't seem to be deprecated and has the added benefit (for now) of being better documented.

**Comment by sowens-csd** [comment]
There is a new example available in version 5.1.0, live now on pub.dev. Thanks for the feedback!
